### ISSUE TITLE
Feature Request: Make all primitive variables a stack

### ISSUE THREAD

**Issue by kp2sushi** [golang/go#14913]
Ok, I had this thought. What if the all the variables could be a stack. The := operator could push the stack or behave normally. Ie, := will create a new variable or will push a stack on an existing variable. = assigns positition 0 on the stack. Subsequent reads pops the stack, unless it is the last element which is persistent across reads.

Just a thought.

**Comment by griesemer** [comment]
This is an interesting idea, but unfortunately the time for language changes has long passed. Also, note that your feature would make it rather difficult to reason about code: Let's say your := is inside a "for" loop
```
var x T
for ... {
   x := foo()
   ...
}
```
Is the stack popped after the for loop or is it as tall as the number of iterations? If it's popped, when is it popped (after each iteration, or at the end of the loop)? And so "Forth"... :-)

In short, I think such a feature would have a hard time in a structured language such as Go.

**Comment by minux** [comment]
What are the use cases for this?

I think this is very confusing as the same identifier can have dramatically
different meaning without assignment.
